app/src/lib/stores/alive-store.ts

```diff
@@ -83,18 +84,33 @@ export class AliveStore {
     this.subscribeToAccounts(accounts)
   }
 
-  private unsubscribeFromAllAccounts() {
+  private async unsubscribeFromAllAccounts() {
+    // Wait until previous (un)subscriptions finish
+    await this.accountSubscriptionPromise
```

Comment:
Yup! From your question, I understand it is a new thing? 🤣 

However, [one small trade-off I learned](https://stackoverflow.com/a/55263084) is that even if the promise is `null`, `await` will yield to the event loop anyway. I don't think we should worry about that but I can do `if (this.accountSubscriptionPromise !== null)` to prevent that if you feel strongly about it.

Reply:
> Yup! From your question, I understand it is a new thing? 🤣

Just an old ghost from my C# days 😄 I'm perfectly happy with us doing so here, I don't think the yield will have any noticeable effects.